File: core/src/main/java/com/triangleleft/flashcards/ui/login/ILoginView.java
package com.triangleleft.flashcards.ui.login;

import com.triangleleft.flashcards.ui.common.view.IView;

import io.reactivex.Observable;

/**
 * View interface for login screen.
 */
public interface ILoginView extends IView {

    void render(LoginViewState viewState);

    Observable<String> logins();

    Observable<String> passwords();

    Observable<Boolean> rememberUserChecks();

    Observable<LoginEvent> loginEvents();
}

File: core/src/main/kotlin/com/triangleleft/flashcards/ui/login/ILoginView.kt
package com.triangleleft.flashcards.ui.login

import com.triangleleft.flashcards.ui.common.view.IView

import io.reactivex.Observable

/**
 * View interface for login screen.
 */
interface ILoginView : IView {

    fun render(viewState: LoginViewState)

    fun logins(): Observable<String>

    fun passwords(): Observable<String>

    fun rememberUserChecks(): Observable<Boolean>

    fun loginEvents(): Observable<LoginEvent>
}
